taichi/lang_util.h

```diff
@@ -199,6 +199,38 @@ class TypedConstant {
   TypedConstant(float64 x) : dt(DataType::f64), val_f64(x) {
   }
 
+  template <typename T>
+  TypedConstant(DataType dt, const T &value) : dt(dt) {
+    if (dt == DataType::f32) {
+      val_f32 = value;
+    } else if (dt == DataType::i32) {
```

Comment:
```suggestion
  if (0) {
#define PER_DATA_TYPE(x) \
  } else if (dt == DataType::x) { \
    val_##x = value;
#include <taichi/inc/data_type.inc.h>
  }
```

Reply:
As I said, we don't have `val_gen` and `val_unknown`, but `taichi/inc/data_type.inc.h` has `gen` and `unknown`.